bank2ynab/ynab_api.py

```diff
@@ -1,29 +1,38 @@
-import requests
 import json
-import configparser
 import logging
+from configparser import DuplicateSectionError, NoSectionError
+
+import requests
 
-import b2y_utilities
+from config_handler import ConfigHandler
 
 # configure our logger
 logging.basicConfig(format="%(levelname)s: %(message)s", level=logging.INFO)
 
 
-class YNAB_API(object):  # in progress (2)
+class YNAB_API:
+    # TODO - revise docstring
+    # TODO - break up class - too many responsibilities
+    #           - ??? what subsets to use
+    # TODO - create API Error class?
+    # TODO - handle transaction creation in DataframeCleaner?
     """
     uses Personal Access Token stored in user_configuration.conf
     (note for devs: be careful not to accidentally share API access token!)
     """
 
-    def __init__(self, config_object, transactions=None):
+    def __init__(
+        self, config_object: ConfigHandler, transactions=None
+    ) -> None:
         self.transactions = []
```

Comment:
Can passing the ConfigHandler be avoided? Can we pass the parameters explicitly instead?